app/src/lib/git/branch.ts

```diff
@@ -73,22 +77,30 @@ export async function deleteBranch(
     await deleteLocalBranch(repository, branch.name)
   }
 
-  const remote = branch.remote
+  const remoteName = branch.remote
 
-  if (includeRemote && remote) {
+  if (includeRemote && remoteName) {
     const networkArguments = await gitNetworkArguments(repository, account)
+    const remoteUrl =
+      (await getRemoteURL(repository, remoteName).catch(err => {
+        // If we can't get the URL then it's very unlikely Git will be able to
+        // either and the push will fail. The URL is only used to resolve the
+        // proxy though so it's not critical.
+        log.error(`Could not resolve remote url for remote ${remoteName}`, err)
+        return null
+      })) || getFallbackUrlForProxyResolve(account, repository)
```

Comment:
just a thought but this _feels_ like a pattern we'll reuse elsewhere over time and might be a nice helper function to pull out sooner than later as we're going through proxy things.

did a quick search and am honestly surprised we don't use `git remote get-url` anywhere else in the codebase currently.

Reply:
Yeah, I was on the fence about whether to extract it early but when I couldn't come up with a name for that method in under 10 seconds I bailed 😛